File: client-clojure/src/sirqul_io_t_platform/specs/account_login_response.clj
(ns sirqul-io-t-platform.specs.account-login-response
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [sirqul-io-t-platform.specs.name-string-value-response :refer :all]
            )
  (:import (java.io File)))


(def account-login-response-data
  {
   (ds/opt :valid) boolean?
   (ds/opt :message) string?
   (ds/opt :version) float?
   (ds/opt :serializeNulls) boolean?
   (ds/opt :startTimeLog) int?
   (ds/opt :errorCode) string?
   (ds/opt :request) (s/coll-of name-string-value-response-spec)
   (ds/opt :accountId) int?
   (ds/opt :locationDisplay) string?
   (ds/opt :display) string?
   (ds/opt :username) string?
   (ds/opt :accountType) string?
   (ds/opt :validated) boolean?
   (ds/opt :returning) string?
   })

(def account-login-response-spec
  (ds/spec
    {:name ::account-login-response
     :spec account-login-response-data}))

File: client-scalaz/src/main/scala/org/openapitools/client/api/AccountLoginResponse.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import AccountLoginResponse._

case class AccountLoginResponse (
  valid: Option[Boolean],
message: Option[String],
version: Option[Double],
serializeNulls: Option[Boolean],
startTimeLog: Option[Long],
errorCode: Option[String],
request: Option[List[NameStringValueResponse]],
accountId: Option[Long],
locationDisplay: Option[String],
display: Option[String],
username: Option[String],
accountType: Option[String],
validated: Option[Boolean],
returning: Option[String])

object AccountLoginResponse {
  import DateTimeCodecs._

  implicit val AccountLoginResponseCodecJson: CodecJson[AccountLoginResponse] = CodecJson.derive[AccountLoginResponse]
  implicit val AccountLoginResponseDecoder: EntityDecoder[AccountLoginResponse] = jsonOf[AccountLoginResponse]
  implicit val AccountLoginResponseEncoder: EntityEncoder[AccountLoginResponse] = jsonEncoderOf[AccountLoginResponse]
}
